pkgs/http_profile/lib/http_profile.dart

```diff
@@ -2,8 +2,208 @@
+      : _timestamp = timestamp,
+        _name = name;
+
+  Map<String, dynamic> _toJson() => <String, dynamic>{
+        'timestamp': _timestamp,
+        'event': _name,
+      };
+}
+
+/// Describes proxy authentication details associated with an HTTP request.
+final class HttpProfileProxyData {
+  final String? _host;
+  final String? _username;
+  final bool? _isDirect;
+  final int? _port;
+
+  HttpProfileProxyData(
+      {String? host, String? username, bool? isDirect, int? port})
+      : _host = host,
+        _username = username,
+        _isDirect = isDirect,
+        _port = port;
+
+  Map<String, dynamic> _toJson() => <String, dynamic>{
+        if (_host != null) 'host': _host,
+        if (_username != null) 'username': _username,
+        if (_isDirect != null) 'isDirect': _isDirect,
+        if (_port != null) 'port': _port,
+      };
+}
+
+/// Describes details about an HTTP request.
+final class HttpProfileRequestData {
+  final Map<String, dynamic> _data;
+
+  final void Function() _updated;
+
+  /// The elements of [connectionInfo] can either be [String]s or [int]s.
+  set connectionInfo(Map<String, dynamic /*String|int*/ > value) {
+    _data['connectionInfo'] = value;
+    _updated();
+  }
+
+  /// The content length of the request, in bytes.
+  set contentLength(int value) {
+    _data['contentLength'] = value;
+    _updated();
+  }
+
+  /// The cookies presented to the server (in the 'cookie' header).
+  set cookies(List<String> value) {
+    _data['cookies'] = value;
+    _updated();
+  }
+
+  /// The error associated with the failed request.
+  set error(String value) {
+    _data['error'] = value;
+    _updated();
+  }
+
+  /// Whether redirects were followed automatically.
+  set followRedirects(bool value) {
+    _data['followRedirects'] = value;
+    _updated();
+  }
+
+  set headers(Map<String, List<String>> value) {
+    _data['headers'] = value;
+    _updated();
+  }
+
+  /// If [followRedirects] is true, this is the maximum number of redirects that
+  /// were followed.
+  set maxRedirects(int value) {
+    _data['maxRedirects'] = value;
+    _updated();
+  }
+
+  /// The requested persistent connection state.
+  set persistentConnection(bool value) {
+    _data['persistentConnection'] = value;
+    _updated();
+  }
+
+  /// Proxy authentication details for the request.
+  set proxyDetails(HttpProfileProxyData value) {
+    _data['proxyDetails'] = value._toJson();
+    _updated();
+  }
+
+  HttpProfileRequestData._(
+      Map<String, dynamic> this._data, void Function() this._updated);
+}
+
+/// Describes details about a response to an HTTP request.
+final class HttpProfileResponseData {
+  final Map<String, dynamic> _data;
+
+  final void Function() _updated;
+
+  /// Records a redirect that the connection went through. The elements of
+  /// [redirect] can either be [String]s or [int]s.
+  void addRedirect(Map<String, dynamic /*String|int*/ > redirect) {
```

Comment:
You could theoretically put arbitrary data in here like with `connectionInfo`, because DevTools just lists out the entries of the map, but I've switched the type of the parameter of `addRedirect` to be a class that only has `statusCode`, `method` and `location` fields. Let me know if you think there's any benefit to switching the parameter type back to `Map<String, dynamic>`.